static/js/apps/eval_rig/feedback_form.tsx

```diff
@@ -37,144 +53,202 @@ export interface EvalInfo {
 export interface Response extends EvalInfo {
   // overall evaluation of all the aspects
   overall: string;
-  userEmail?: string;
 }
 
-export interface FeedbackFormProps {
-  queryId: string;
-  callId: string;
-  evalInfo: EvalInfo;
-}
+const emptyResponse = {
+  overall: "",
+  question: "",
+  llmStat: "",
+  dcResponse: "",
+  dcStat: "",
+};
+
+const FeedbackForm = forwardRef((_, ref) => {
+  const { allCall, doc, sheetId, userEmail } = useContext(AppContext);
+  const { sessionQueryId, sessionCallId, setSessionCallId } =
+    useContext(SessionContext);
 
-export function FeedbackForm(props: FeedbackFormProps): JSX.Element {
-  const { doc, sheetId, userEmail } = useContext(AppContext);
-  const [response, setResponse] = useState<Response>(initialResponse);
-  const [completed, SetCompleted] = useState(false);
+  const [evalInfo, setEvalInfo] = useState<EvalInfo | null>(null);
+  const [response, setResponse] = useState<Response>(emptyResponse);
+  const [status, setStatus] = useState<FormStatus>(null);
 
   useEffect(() => {
-    getCallData(sheetId, props.queryId, props.callId).then((data) => {
+    getCallData(sheetId, sessionQueryId, sessionCallId).then((data) => {
       if (data) {
         setResponse(data as Response);
-        SetCompleted(true);
+        setStatus(FormStatus.Submitted);
       } else {
-        setResponse(initialResponse);
-        SetCompleted(false);
+        setResponse(emptyResponse);
+        setStatus(FormStatus.NotStarted);
       }
     });
-  }, [sheetId, props.queryId, props.callId]);
+  }, [sheetId, sessionQueryId, sessionCallId]);
+
+  useEffect(() => {
+    const sheet = doc.sheetsByTitle[DC_CALL_SHEET];
+    if (!(sessionQueryId in allCall)) {
+      setEvalInfo(null);
+      return;
+    }
+    const rowIdx = allCall[sessionQueryId][sessionCallId];
+    sheet.getRows({ offset: rowIdx - 1, limit: 1 }).then((rows) => {
+      const row = rows[0];
+      if (row) {
+        setEvalInfo({
+          dcResponse: row.get(DC_RESPONSE_COL),
+          dcStat: row.get(DC_STAT_COL),
+          llmStat: row.get(LLM_STAT_COL),
+          question: row.get(DC_QUESTION_COL),
+        });
+      }
+    });
+  }, [doc, allCall, sessionQueryId, sessionCallId, setSessionCallId]);
 
   const handleChange = (event) => {
     const { name, value } = event.target;
-    setResponse((prevState) => ({
-      ...prevState,
-      [name]: value,
-    }));
+    setResponse((prevState) => {
+      const newState = {
+        ...prevState,
+        [name]: value,
+      };
+      let tmpStatus = FormStatus.Completed;
+      for (const value of Object.values(newState)) {
+        if (value === "") {
+          tmpStatus = FormStatus.InProgress;
+          break;
+        }
+      }
+      setStatus(tmpStatus);
```

Comment:
why is this a tmpStatus if it will be set as the status eventually?

Reply:
trying not to use the "status" variable name..